include/mbedtls/camellia.h

```diff
@@ -221,13 +221,17 @@ int mbedtls_camellia_crypt_ctr( mbedtls_camellia_context *ctx,
 extern "C" {
 #endif
 
+#if defined(MBEDTLS_SELF_TEST)
+
 /**
  * \brief          Checkup routine
  *
  * \return         0 if successful, or 1 if the test failed
  */
 int mbedtls_camellia_self_test( int verbose );
 
+#endif /*MBEDTLS_SELF_TEST*/
```

Comment:
I thought about it, but then it is risky if some will add new public APIs which should not be replaced by alternative implementation, and miss that it's surrounded by the extern "C" {} block